clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,390 @@
+            write_to_tsv(participants, participant_file)
+
+    for participant_id, data_frame in sessions.groupby(["participant_id"]):
+        session = data_frame.droplevel(
+            ["participant_id", "modality", "bids_filename"]
+        ).drop_duplicates()
+
+        session_filepath = to / participant_id / f"{participant_id}_sessions.tsv"
+        with fs.open(session_filepath, "w") as sessions_file:
+            write_to_tsv(session, sessions_file)
+    scans = scans.set_index(["bids_full_path"], verify_integrity=True)
+    for bids_full_path, metadata in scans.iterrows():
+        if metadata["modality_num"] != "20217" and metadata["modality_num"] != "20225":
+            copy_file_to_bids(
+                zipfile=str(dataset_directory) + "/" + metadata["source_zipfile"],
+                filenames=[metadata["source_filename"]] + metadata["sidecars"],
+                bids_path=to / bids_full_path,
+            )
+        else:
+            convert_dicom_to_nifti(
+                zipfiles=str(dataset_directory) + "/" + metadata["source_zipfile"],
+                bids_path=to / bids_full_path,
+            )
+    return
+
+
+def copy_file_to_bids(zipfile: str, filenames: List[str], bids_path: str) -> None:
+    """Install the requested files in the BIDS  dataset."""
+    import fsspec
+
+    fo = fsspec.open(zipfile)
+    fs = fsspec.filesystem("zip", fo=fo)
+    for filename in filenames:
+        if fs.exists(filename):
+            bids_path_extension = str(bids_path) + "." + (filename.split(".", 1)[1])
+            with fsspec.open(bids_path_extension, mode="wb") as f:
+                f.write(fs.cat(filename))
+
+
+def convert_dicom_to_nifti(zipfiles: str, bids_path: str) -> None:
+    """Install the requested files in the BIDS  dataset.
+    First, the dicom is extracted in a temporary directory
+    Second, the dicom extracted is converted in the right place using dcm2niix"""
+    import os
+    import subprocess
+    import tempfile
+    import zipfile
+    from pathlib import Path
+
+    zf = zipfile.ZipFile(zipfiles)
+    try:
+        os.makedirs(Path(bids_path).parent)
+    except OSError:
+        # Folder already created with previous instance
+        pass
+    with tempfile.TemporaryDirectory() as tempdir:
+        zf.extractall(tempdir)
+        command = [
+            "dcm2niix",
+            "-w",
+            "0",
+            "-f",
+            Path(bids_path).name,
+            "-o",
+            Path(bids_path).parent,
+        ]
+        command += ["-9", "-z", "y"]
+        command += ["-b", "y", "-ba", "y"]
+        command += [tempdir]
+        subprocess.run(command)
+    return
+
+
+def select_session(x: int) -> int:
+    if x["source_session_number"] == "2":
+        return x.age_when_attended_assessment_centre_f21003_2_0
+    elif x["source_session_number"] == "3":
+        return x.age_when_attended_assessment_centre_f21003_3_0
```

Comment:
Static analysis complains and rightfully so. `x` is definitely not an int here.

Could this be made simpler? At the end of the day all we need is a mapping:

```python
{
    "2": "age_when_attended_assessment_centre_f21003_2_0",
    "3": "age_when_attended_assessment_centre_f21003_3_0",
}
```

or a function:

```python
def get_age_column(session: int) -> str:
    return f"age_when_attended_assessment_centre_f21003_{session}_0"
```